### ISSUE TITLE
> library(Seurat) zsh: killed     R -e "library(Seurat)"

### ISSUE THREAD

**Issue by pineappleburger** [satijalab/seurat#3577]
My original problem was that `rmarkdown::render("Debugging/Untitled.Rmd")` in the RStudio console works and produces the PDF file – But doesn't do so if run in a new R session in the terminal  or over the RStudio knit button.

Reprex:
````
---
title: "Untitled"
output: pdf_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```


```{r loadSeurat}
library(Seurat)
```
````

Output when run when knitted with `R -e "rmarkdown::render('Untitled.Rmd')"`:
````
> rmarkdown::render('Untitled.Rmd')


processing file: Untitled.Rmd
  |..................                                                    |  25%
  ordinary text without R code

  |...................................                                   |  50%
label: setup (with options) 
List of 1
 $ include: logi FALSE

  |....................................................                  |  75%
  ordinary text without R code

  |......................................................................| 100%
label: loadSeurat
zsh: killed     R -e "rmarkdown::render('Untitled.Rmd')"
````

I noticed the same goes just for the command `library(Seurat)`. It works just fine in the RStudio console, but running `R -e "library(Seurat)"`in terminal fails: `zsh: killed R -e "library(Seurat)`.

Would anyone happen to have an idea on this?

_Originally posted by @pineappleburger in closed issue https://github.com/satijalab/seurat/issues/2845#issuecomment-701445359_

Session Info in Terminal:

```
R version 4.0.2 (2020-06-22)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Catalina 10.15.6

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRblas.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRlapack.dylib

locale:
[1] de_DE.UTF-8/de_DE.UTF-8/de_DE.UTF-8/C/de_DE.UTF-8/de_DE.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

loaded via a namespace (and not attached):
[1] compiler_4.0.2
```

Loading Seurat in the R console in RStudio works fine:

```
> library("Seurat")
> sessionInfo()
R version 4.0.2 (2020-06-22)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Catalina 10.15.6

Matrix products: default
BLAS:   /System/Library/Frameworks/Accelerate.framework/Versions/A/Frameworks/vecLib.framework/Versions/A/libBLAS.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRlapack.dylib

locale:
[1] de_DE.UTF-8/de_DE.UTF-8/de_DE.UTF-8/C/de_DE.UTF-8/de_DE.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] Seurat_3.2.2

loaded via a namespace (and not attached):
 [1] nlme_3.1-149          matrixStats_0.57.0    RcppAnnoy_0.0.16      RColorBrewer_1.1-2    httr_1.4.2            sctransform_0.3       tools_4.0.2           R6_2.4.1              irlba_2.3.3          
[10] rpart_4.1-15          KernSmooth_2.23-17    uwot_0.1.8            lazyeval_0.2.2        mgcv_1.8-33           colorspace_1.4-1      tidyselect_1.1.0      gridExtra_2.3         compiler_4.0.2       
[19] plotly_4.9.2.1        scales_1.1.1          lmtest_0.9-38         spatstat.data_1.4-3   ggridges_0.5.2        pbapply_1.4-3         spatstat_1.64-1       goftest_1.2-2         stringr_1.4.0        
[28] digest_0.6.25         spatstat.utils_1.17-0 pkgconfig_2.0.3       htmltools_0.5.0       fastmap_1.0.1         htmlwidgets_1.5.2     rlang_0.4.7           rstudioapi_0.11       shiny_1.5.0          
[37] generics_0.0.2        zoo_1.8-8             jsonlite_1.7.1        ica_1.0-2             dplyr_1.0.2           magrittr_1.5          patchwork_1.0.1       Matrix_1.2-18         Rcpp_1.0.5           
[46] munsell_0.5.0         abind_1.4-5           reticulate_1.16       lifecycle_0.2.0       stringi_1.5.3         MASS_7.3-53           Rtsne_0.15            plyr_1.8.6            grid_4.0.2           
[55] parallel_4.0.2        listenv_0.8.0         promises_1.1.1        ggrepel_0.8.2         crayon_1.3.4          miniUI_0.1.1.1        deldir_0.1-29         lattice_0.20-41       cowplot_1.1.0        
[64] splines_4.0.2         tensor_1.5            pillar_1.4.6          igraph_1.2.6          future.apply_1.6.0    reshape2_1.4.4        codetools_0.2-16      leiden_0.3.3          glue_1.4.2           
[73] data.table_1.13.0     png_0.1-7             vctrs_0.3.4           httpuv_1.5.4          gtable_0.3.0          RANN_2.6.1            purrr_0.3.4           polyclip_1.10-0       tidyr_1.1.2          
[82] future_1.19.1         ggplot2_3.3.2         rsvd_1.0.3            mime_0.9              xtable_1.8-4          later_1.1.0.1         survival_3.2-7        viridisLite_0.3.0     tibble_3.0.3         
[91] cluster_2.1.0         globals_0.13.0        fitdistrplus_1.1-1    ellipsis_0.3.1        ROCR_1.0-11   
```

I tried to reinstall all installed packages, unfortunately the problem persists.
```
> BiocManager::valid()
[1] TRUE
```

**Comment by timoast** [comment]
I wasn't able to reproduce this on macOS 10.15.7 with R 4.0.2 (also using zsh)

Here's my sessionInfo after loading Seurat:


```
R version 4.0.2 (2020-06-22)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Catalina 10.15.7

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRblas.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRlapack.dylib

locale:
[1] en_AU.UTF-8/en_AU.UTF-8/en_AU.UTF-8/C/en_AU.UTF-8/en_AU.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base

other attached packages:
[1] Seurat_3.2.2

loaded via a namespace (and not attached):
 [1] httr_1.4.2            tidyr_1.1.2           jsonlite_1.7.1
 [4] viridisLite_0.3.0     splines_4.0.2         leiden_0.3.3
 [7] shiny_1.5.0           ggrepel_0.8.2         globals_0.13.0
[10] pillar_1.4.6          lattice_0.20-41       glue_1.4.2
[13] reticulate_1.16       digest_0.6.25         polyclip_1.10-0
[16] RColorBrewer_1.1-2    promises_1.1.1        colorspace_1.4-1
[19] cowplot_1.1.0         htmltools_0.5.0       httpuv_1.5.4
[22] Matrix_1.2-18         plyr_1.8.6            pkgconfig_2.0.3
[25] listenv_0.8.0         purrr_0.3.4           xtable_1.8-4
[28] patchwork_1.0.1       scales_1.1.1          RANN_2.6.1
[31] tensor_1.5            later_1.1.0.1         Rtsne_0.15
[34] spatstat.utils_1.17-0 tibble_3.0.3          mgcv_1.8-33
[37] generics_0.0.2        ggplot2_3.3.2         ellipsis_0.3.1
[40] ROCR_1.0-11           pbapply_1.4-3         lazyeval_0.2.2
[43] deldir_0.1-29         survival_3.2-3        magrittr_1.5
[46] crayon_1.3.4          mime_0.9              future_1.19.1
[49] nlme_3.1-149          MASS_7.3-53           ica_1.0-2
[52] tools_4.0.2           fitdistrplus_1.1-1    data.table_1.13.0
[55] lifecycle_0.2.0       matrixStats_0.57.0    stringr_1.4.0
[58] plotly_4.9.2.1        munsell_0.5.0         cluster_2.1.0
[61] irlba_2.3.3           compiler_4.0.2        rsvd_1.0.3
[64] rlang_0.4.7           grid_4.0.2            ggridges_0.5.2
[67] goftest_1.2-2         RcppAnnoy_0.0.16      rappdirs_0.3.1
[70] htmlwidgets_1.5.1     igraph_1.2.5          miniUI_0.1.1.1
[73] gtable_0.3.0          codetools_0.2-16      abind_1.4-5
[76] reshape2_1.4.4        R6_2.4.1              gridExtra_2.3
[79] zoo_1.8-8             dplyr_1.0.2           uwot_0.1.8
[82] fastmap_1.0.1         future.apply_1.6.0    KernSmooth_2.23-17
[85] spatstat.data_1.4-3   stringi_1.5.3         spatstat_1.64-1
[88] parallel_4.0.2        Rcpp_1.0.5            rpart_4.1-15
[91] vctrs_0.3.4           sctransform_0.3       png_0.1-7
[94] tidyselect_1.1.0      lmtest_0.9-38
```

**Comment by pineappleburger** [comment]
@timoast, thank you for your fast reply! I tried to load the packages you kindly provided a list of one by one and found the problem: the leiden packages failed to load because it loaded the wrong python version (without numpy/leidenalg/python-igraph). It's my own fault really; I do use those packages, but only in a conda environment.

My problem ended up being somewhat off-topic, thanks for your help! 

Steps:
<details>
```
Jo@Workspace ~ % . /Users/Jo/anaconda3/bin/activate && conda activate /Users/Jo/anaconda3/envs/myenv;
(myenv) Jo@Workspace ~ % where python
/Users/Jo/anaconda3/envs/myenv/bin/python
/usr/bin/python
(myenv) Jo@Workspace ~ % cd PATH/TO/PROJECT
(myenv) Jo@Workspace PROJECT % touch .Rprofile
(myenv) Jo@Workspace PROJECT % nano .Rprofile
```

Where I added `Sys.setenv(RETICULATE_PYTHON = "Users/Jo/anaconda3/envs/myenv/bin/python")`.
</details>
